File: LeetCode/Easy/3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        for (; ; n++) {
            int p = 1, k = n;
            for (; k > 0; k /= 10) p *= (k % 10);
            if (p % t == 0) return n;
        }
    }
};

File: LeetCode/Easy/3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {
        for (int k, p = 1; ; n++, p = 1) {
            for (k = n; k > 0; k /= 10) p *= (k % 10);
            if (p % t == 0) return n;
        }
    }
}